[webkitpy] Change Simulator Timeout logic

robert-jenner opened the pull request:
#### 1b31c70805304cd5257c6d70332632e8e86161d5
<pre>
[webkitpy] Change Simulator Timeout logic
<a href="https://rdar.apple.com/166828504">rdar://166828504</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=304449">https://bugs.webkit.org/show_bug.cgi?id=304449</a>

Reviewed by Jonathan Bedard.

Change simulator timeout logic for an increase in complete test runs.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDevice.is_usable):
(SimulatedDevice.launch_app):

Canonical link: <a href="https://commits.webkit.org/304762@main">https://commits.webkit.org/304762@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f9a3f1dcf4147ac9c595267c89ace44fd7f805ae

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136478 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8835 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47758 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144191 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/89449 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6372f643-b286-47a4-afa2-6c7ca4aa0e95) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/138350 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9522 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8679 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/104363 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/89449 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/72cb4252-4255-475f-8a72-feb5f37bfade) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/139423 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6949 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/122284 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85198 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/9efb1fd6-66e5-4d7c-a6e1-dbbf31bc9d13) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/135825 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/6593 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/4254 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4784 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/115894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40489 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146940 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8517 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41058 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/112704 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8534 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/7157 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113047 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28700 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/6525 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118593 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/62505 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8565 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/36645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/8284 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/8505 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/8357 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

bls1999 commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
```diff
@@ -646,7 +646,7 @@ def is_usable(self, force_update=False):
 
         try:
-            with Timeout(seconds=30, patch=False):
+            with Timeout(seconds=300, patch=False):
```
This can definitely be 30 seconds -- even under load, terminate shouldn't take more than a few seconds at most.

bls1999 reviewed (approved):
[unofficial] r=me

One non-blocking comment. We should really audit all the simulator timeouts as many are remnants of a bygone era…

JonWBedard commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
Instead of adjusting the timeouts, let's attack this line. We should:
- Return `False` instead of `None`
- Log "Exceeded timeout while running simctl terminate, not simulator frames executing"

robert-jenner committed:
[webkitpy] Change Simulator Timeout logic
rdar://166828504
https://bugs.webkit.org/show_bug.cgi?id=304449

Reviewed by Jonathan Bedard.

Change simulator timeout logic for an increase in complete test runs.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDevice.is_usable):
(SimulatedDevice.launch_app):

Canonical link: https://commits.webkit.org/304762@main
